src/compiler/scala/tools/nsc/transform/Mixin.scala

```diff
@@ -549,13 +541,13 @@ abstract class Mixin extends Transform with ast.TreeDSL with AccessorSynthesis {
       import accessorSynth._
 
       // for all symbols `sym` in the class definition, which are mixed in by mixinTraitMembers
-      for (sym <- clazz.info.decls ; if sym hasFlag MIXEDIN) {
+      for (sym <- clazz.info.decls if sym.hasFlag(MIXEDIN))
         // if current class is a trait, add an abstract method for accessor `sym`
         // ditto for a super accessor (will get an RHS in completeSuperAccessor)
         if (clazz.isTrait || sym.isSuperAccessor) addDefDef(sym)
         // implement methods mixed in from a supertrait (the symbols were created by mixinTraitMembers)
-        else if (sym.hasFlag(ACCESSOR) && !sym.hasFlag(DEFERRED)) {
-          assert(sym hasFlag (PARAMACCESSOR), s"mixed in $sym from $clazz is not param?!?")
+        else if (sym.hasFlag(ACCESSOR) && sym.hasNoFlags(DEFERRED|LAZY)) {
```

Comment:
This doesn't skip only the assertion, but also `addDedDef`. Not sure if that's correct?

Reply:
I ran out of midnight before I learned what is going on. Adriaan left comments about lazy vals from fields work, which are on my reading list. Mixin used to do work for lazy vals before then?

Reply:
Plenty of time to add tests before 2.13.13.

Reply:
berühmte letzte Worte.